client/signup/config/test/index.js

```diff
@@ -17,4 +19,57 @@ describe( 'index', () => {
 			);
 		}
 	} );
+
+	test( 'All steps should have the same dependencies provided', () => {
+		const stepModuleMap = getStepModuleMap();
+		const moduleStepMap = {};
+		Object.entries( stepModuleMap ).forEach( ( [ step, module ] ) => {
+			if ( ! moduleStepMap[ module ] ) {
+				moduleStepMap[ module ] = [];
+			}
+			moduleStepMap[ module ].push( step );
+		} );
+		const allStepDefinitions = generateSteps();
+
+		Object.values( moduleStepMap ).forEach( ( moduleRelatedSteps ) => {
+			const allDependenciesProvidedForModule = new Set();
```

Comment:
Do we need to create this if `moduleRelatedSteps.length` is `0`? Perhaps it should be moved inside the `if`?